lib/livebook_web/live/session_live/secrets_component.ex

```diff
@@ -63,21 +63,49 @@ defmodule LivebookWeb.SessionLive.SecretsComponent do
           </div>
         </div>
       </.form>
+      <%= if @ref do %>
+        <h3 class="text-2xl font-semibold text-gray-800">
+          Select secret
+        </h3>
+        <.form
+          let={_}
+          for={:secrets}
+          phx-submit="save_secret"
+          phx-change="select_secret"
+          phx-target={@myself}
+        >
+          <.select name="secret" selected={@current_secret} options={secret_options(@secrets)} />
+        </.form>
+      <% end %>
     </div>
     """
   end
 
   @impl true
   def handle_event("save", %{"data" => data}, socket) do
+    secret_label = String.upcase(data["label"])
+
     if data_errors(data) == [] do
-      secret = %{label: String.upcase(data["label"]), value: data["value"]}
+      secret = %{label: secret_label, value: data["value"]}
       Livebook.Session.put_secret(socket.assigns.session.pid, secret)
-      {:noreply, push_patch(socket, to: socket.assigns.return_to)}
+
+      {:noreply,
+       socket |> push_patch(to: socket.assigns.return_to) |> maybe_update_secret(secret_label)}
     else
       {:noreply, assign(socket, data: data)}
     end
   end
 
+  def handle_event("select_secret", %{"secret" => secret_label}, socket) do
+    {:noreply,
+     socket
+     |> push_patch(to: socket.assigns.return_to)
+     |> push_event("secret_selected", %{
+       select_secret_ref: socket.assigns.ref,
+       secret_label: secret_label
+     })}
```

Comment:
```suggestion
     |> maybe_update_secret(secret_label)}
```

and I'd rename the function to `push_secret_selected` :)